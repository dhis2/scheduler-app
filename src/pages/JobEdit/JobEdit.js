import React from 'react'
import { Card, IconInfo16 } from '@dhis2/ui'
import PropTypes from 'prop-types'
import i18n from '@dhis2/d2-i18n'
import { JobEditFormContainer } from '../../components/Forms'
import { JobDetails } from '../../components/JobDetails'
import styles from './JobEdit.module.css'

const infoLink =
    'https://docs.dhis2.org/en/use/user-guides/dhis-core-version-236/maintaining-the-system/scheduling.html'

const JobEdit = ({ job }) => {
    const { name, created, lastExecutedStatus, lastExecuted } = job

    return (
        <React.Fragment>
            <header className={styles.pageHeader}>
                <h2 className={styles.pageHeaderTitle}>
                    {i18n.t('Job: {{ name }}', {
                        name,
                        nsSeparator: '>',
                    })}
                </h2>
            </header>
            <Card className={styles.card}>
                <header className={styles.cardHeader}>
                    <h3 className={styles.cardHeaderTitle}>
                        {i18n.t('Configuration')}
                    </h3>
                    <a
                        href={infoLink}
                        className={styles.cardHeaderLink}
                        target="_blank"
                        rel="noopener noreferrer"
                    >
                        <span className={styles.cardHeaderInfo}>
                            <IconInfo16 />
                        </span>
                        {i18n.t('About job configuration')}
                    </a>
                </header>
                <div className={styles.jobDetails}>
                    <JobDetails
                        created={created}
                        lastExecutedStatus={lastExecutedStatus}
                        lastExecuted={lastExecuted}
                    />
                </div>
                <JobEditFormContainer job={job} />
            </Card>
        </React.Fragment>
    )
}

const { shape, string } = PropTypes

JobEdit.propTypes = {
    job: shape({
        name: string.isRequired,
        created: string.isRequired,
        lastExecutedStatus: string.isRequired,
        lastExecuted: string.isRequired,
    }).isRequired,
}

export default JobEdit
