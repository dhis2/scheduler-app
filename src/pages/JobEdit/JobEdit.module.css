.pageHeader {
    margin-bottom: var(--spacers-dp32);
}

.pageHeaderButton {
    margin-bottom: var(--spacers-dp16);
}

.pageHeaderTitle {
    font-size: 18px;
    font-weight: 400;
    line-height: 1;
    margin: 0;
}

.card {
    padding: var(--spacers-dp16);
}

.cardHeader {
    align-items: center;
    display: inline-flex;
    margin-bottom: var(--spacers-dp16);
}

.cardHeaderTitle {
    font-size: 16px;
    font-weight: 500;
    margin: 0 var(--spacers-dp8) 0 0;
}

.cardHeaderInfo {
    margin-right: var(--spacers-dp4);
}

.cardHeaderLink {
    align-items: center;
    display: flex;
    font-size: 12px;
    color: var(--colors-grey600);
    fill: var(--colors-grey600);
}

.jobDetails {
    float: right;
}
