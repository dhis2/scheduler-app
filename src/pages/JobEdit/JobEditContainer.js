import React, { useState } from 'react'
import { CircularLoader, Layer, CenteredContent } from '@dhis2/ui'
import { useParams } from 'react-router-dom'
import { useDataQuery } from '@dhis2/app-runtime'
import JobEdit from './JobEdit'

const query = {
    job: {
        resource: 'jobConfigurations',
        id: /* istanbul ignore next */ ({ id }) => id,
        params: {
            paging: false,
        },
    },
}

const JobEditContainer = () => {
    const [isPristine, setIsPristine] = useState(true)
    const { id } = useParams()
    const { loading, error, data } = useDataQuery(query, { variables: { id } })

    if (loading) {
        return (
            <Layer>
                <CenteredContent>
                    <CircularLoader />
                </CenteredContent>
            </Layer>
        )
    }

    if (error) {
        /**
         * The app can't continue if this fails, because it doesn't
         * have the job data, so throw the error.
         */
        throw error
    }

    return (
        <JobEdit
            isPristine={isPristine}
            setIsPristine={setIsPristine}
            name={data.job.name}
        />
    )
}

export default JobEditContainer
