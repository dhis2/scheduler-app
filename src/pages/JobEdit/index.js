import JobEdit from './JobEdit'

export default JobEdit
