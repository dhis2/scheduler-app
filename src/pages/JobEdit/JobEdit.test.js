import React from 'react'
import { shallow } from 'enzyme'
import JobEdit from './JobEdit'

describe('<JobEdit>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<JobEdit />)

        expect(wrapper).toMatchSnapshot()
    })
})
