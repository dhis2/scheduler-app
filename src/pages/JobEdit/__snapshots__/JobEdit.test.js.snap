// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobEdit> renders correctly 1`] = `
<div>
  Edit job
</div>
`;
